data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java

```diff
@@ -29,30 +29,30 @@
 @NoArgsConstructor
 public class EnvironmentId {
 
-  private static final String UNDERSCORE = "_";
+  private static final String JOINER = "/";
```

Comment:
Actually AWS ARN doesn't forbid to use "/" in the resource even slash is used for ARN format like `arn:partition:service:region:account-id:resourcetype/resource`, you may see some services use '/' in their resource ID.

But inside ECS,   we don't allow '/' in the name, e..g in the cluster name, it says "The name of your cluster. If you do not specify a name for your cluster, you create a cluster named default. Up to 255 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed."

I think use slash is good idea so far.  The UNDERSCORE is definitely buggy as cluster name allowed to contain underscore.

Reply:
Should we add this message in Blox front end as well, so people know that '/' is not allowed in names? 
We can do this in another sim...not necessarily in this PR.

Reply:
I don't think we allow it. We can follow the same name requirement as cluster, like "Up to 255 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed.". In this way,  it basically doesn't allow slash.

Reply:
That's true. My suggestion is that we should output this similar message to the user when they try to use names with slash :-D

Reply:
We are getting these cluster names only from ECS, so there is no need for Blox to validate the cluster name. But yeah, we definitely need to add validation and return some message to indicate the invalid parameters when customer call createEnvironment in FE. Will check and address it in future PR.